### ISSUE TITLE
Stacks deployed with versions earlier than 1.93 cause TypeError: Cannot read properties of undefined (reading 'split')

### ISSUE THREAD

**Issue by TimQuist** [cdklabs/awscdk-v1-stack-finder#8]
Hi,

Using this tool to scan for V1 CDK stacks. Works fine on most of my accounts, however on one of the accounts I am getting the following error:

```bash
TypeError: Cannot read properties of undefined (reading 'split')
    at findV1Stacks (.npm/_npx/1a70b0cb204f1c57/node_modules/awscdk-v1-stack-finder/lib/index.js:149:85)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async processListStacksResponse (.npm/_npx/1a70b0cb204f1c57/node_modules/awscdk-v1-stack-finder/lib/index.js:122:5)
    at async find (.npm/_npx/1a70b0cb204f1c57/node_modules/awscdk-v1-stack-finder/lib/index.js:94:9).
```

It's going wrong in the following function:
```js
async function findV1Stacks(response, cloudformation, region, stacks) {
    var _a;
    for (const stack of (_a = response.StackSummaries) !== null && _a !== void 0 ? _a : []) {
        const getTemplateResponse = await cloudformation.getTemplate({
            StackName: stack.StackName,
        }).promise().catch((error) => console.error(`${region}: Failed to get template for stack: ${stack.StackName}. Error: ${error}`));
        var body;
        try {
            body = JSON.parse(getTemplateResponse.TemplateBody);
        }
        catch (err) {
            body = YAML.parse(getTemplateResponse.TemplateBody);
        }
        if (body.Resources.CDKMetadata) {
            const buf = Buffer.from(body.Resources.CDKMetadata.Properties.Analytics.split(':').splice(2)[0], 'base64');
            const analyticsString = zlib.gunzipSync(buf).toString();
            const majorVersion = analyticsString.slice(0, 1);
            if (majorVersion === '1') {
                stacks.push(`name: ${stack.StackName} | id: ${stack.StackId}`);
            }
        }
    }
}
```

Something is causing `body.Resources.CDKMetadata.Properties.Analytics` to be `undefined`.

- I am assuming a role with all permissions and no explicit denies using the `aws sts assume-role` api.
- I am using the env variable to set the credentials after.
```bash
export AWS_ACCESS_KEY_ID=...
export AWS_SECRET_ACCESS_KEY=...
export AWS_SESSION_TOKEN=...
```
- I verified the account has `CDKToolkit` stacks.
-  Node.js version v16.14.2, npm version 8.5.0

**Comment by madeline-k** [comment]
Can you share the templates of the CloudFormation stacks in that account? Or just the CDKMetadata resource snippets from the templates? 

In the meanwhile, I can add better error handling/messaging for this scenario.

**Comment by madeline-k** [comment]
Thanks for the quick response @TimQuist! This should be enough information to address the issue. Your `guard-duty-master-stack` was created with cdk version 1.82.0, and the v1-stack-finder does not account for how the CDKMetadata was formatted in versions that far back.

**Comment by TimQuist** [comment]
@madeline-k 

Great!

Some context:
I figured out that we did in fact not roll out to this account during our deploys. Since it only contains guard-duty and pretty much never changes this went unnoticed. In our CDK projects the versions were upgraded multiple times already, but just never rolled out to this account. This is why it was still on a pretty old version. I am glad to have noticed this so now we can include this account in the deploys 😄
